shared/macros-oval.jinja

```diff
@@ -0,0 +1,199 @@
+#}}
+{{%- macro oval_line_in_file_criterion(path='', parameter='', default_pass=false) -%}}
+{{%- set prefix_text = "Check the" -%}}
+{{%- set suffix_id = "" -%}}
+{{%- if default_pass %}}
+{{%- set suffix_id = "_default_pass" -%}}
+{{%- set prefix_text = prefix_text + " absence of" -%}}
+{{%- endif %}}
+      <criterion comment="{{{ prefix_text }}} {{{ parameter }}} in {{{ path }}}"
+      test_ref="test_{{{ rule_id }}}{{{ suffix_id }}}" />
+{{%- endmacro %}}
+
+{{#
+  Macro to define the OVAL test to be constructed (Test definition).
+  This function can take three parameters:
+    - path (String): Path to the configuration file to be checked.
+    - parameter (String): The parameter to be checked in the configuration file.
+    - default_pass (boolean): Implies that the value for the parameter is the default value. If set, the check will pass even if the parameter is not set in the configuration file.
+#}}
+{{%- macro oval_line_in_file_test(path='', parameter='', default_pass=false) -%}}
+{{%- if default_pass %}}
+{{%- set check_existence = "none_exist" -%}}
+{{%- set prefix_text = "value" -%}}
+{{%- set suffix_id = "_default_pass" -%}}
+{{%- else %}}
+{{%- set check_existence = "all_exist" -%}}
+{{%- set prefix_text = "absence" -%}}
+{{%- set suffix_id = "" -%}}
+{{%- endif %}}
+  <ind:textfilecontent54_test check="all" check_existence="{{{ check_existence }}}"
+  comment="tests the {{{ prefix_text }}} of {{{ parameter }}} setting in the {{{ path }}} file"
+  id="test_{{{ rule_id }}}{{{ suffix_id }}}" version="1">
+  <ind:object object_ref="obj_{{{ rule_id }}}{{{ suffix_id }}}" />
+  {{%- if not default_pass %}}
+  <ind:state state_ref="state_{{{ rule_id }}}{{{ suffix_id }}}" />
+  {{%- endif %}}
+  </ind:textfilecontent54_test>
+{{%- endmacro %}}
+
+{{#
+  Macro to check if a parameter in a configuration file is set (Object definition).
+  This function can take five parameters:
+    - path (String): Path to the configuration file to be checked.
+    - prefix_regex (String): Regular expression to be used in the beginning of the OVAL text file content check.
+    - parameter (String): The parameter to be checked in the configuration file.
+    - separator_regex (String): Regular expression to be used as the separator of parameter and value in a configuration file. If spaces are allowed, this should be included in the regular expression.
+    - default_pass (boolean): Implies that the value for the parameter is the default value. If set, the check will pass even if the parameter is not set in the configuration file.
+#}}
+{{%- macro oval_line_in_file_object(path='', prefix_regex='^[\s]*', parameter='', separator_regex='[\s]+', default_pass='') -%}}
+{{%- if default_pass %}}
+{{%- set suffix_id = "_default_pass" -%}}
+{{%- set regex = prefix_regex+parameter+separator_regex -%}}
```

Comment:
I would add a comment here to explain that there is no need for a capture group when `default_pass=true`, because when `default_pass=true` the test doesn't have a state and it checks for the absence of the `parameter` in config file.